12-rest/classworks/Volodkova_Anna/index.js

```diff
@@ -0,0 +1,114 @@
+const koa = require('koa');
+const bodyParser = require('koa-bodyparser');
+const Router = require('koa-router');
+const crypto = require('crypto');
+const jsonwebtoken = require('jsonwebtoken');
+const jwt = require('koa-jwt');
+const  fs = require('fs');
+const fetch = require('node-fetch');
+
+const users = JSON.parse(fs.readFileSync('./users.json'));
+const taglines = JSON.parse(fs.readFileSync('./taglines.json'));
+const SECRETKEY = 'shhhhh';
+
+const app = new koa();
+const publicRouter = new Router();
+const privateRouter = new Router();
+
+publicRouter
+  .post('/signUp', async (ctx) => {
+    console.log('signUp body', ctx.request.body);
+    const logIn = ctx.request.body.logIn;
+    const password = ctx.request.body.password;
+    const passwordHash = myCripto(password);
+    const userObj = {
+      logIn: logIn,
+      password: passwordHash,
+    }
+    users.push(userObj);
+    fs.writeFileSync('./users.json', JSON.stringify(users, null, 2));
+    ctx.status = 204;
+    ctx.body = JSON.stringify({});
+  })
+  .post('/signIn', (ctx) => {
+    console.log('signIn body', ctx.request.body);
+    const logIn = ctx.request.body.logIn;
+    const password = ctx.request.body.password;
+    const passwordHash = myCripto(password);
+
+    const userObj = {
+      logIn: logIn,
+      password: passwordHash,
+    }
+    const user = users.find((u) => {
+      return userObj.logIn === u.logIn;
+    });
+
+    const isLogIn = user.password === passwordHash;
```

Comment:
why do we find user with such login and then check his password if we can do it at the same time